eng/targets/Settings.props

```diff
@@ -60,6 +60,9 @@
 
     <!-- TODO --> 
     <_SkipUpgradeNetAnalyzersNuGetWarning>true</_SkipUpgradeNetAnalyzersNuGetWarning>
+
+    <!-- https://github.com/dotnet/msbuild/issues/10306 -->
+    <CoreCompileDependsOn>$(CoreCompileDependsOn);ResolveKeySource</CoreCompileDependsOn> 
```

Comment:
This will solve the immediate issue but I'm worried about other similar target dependencies. No objection to checking it in but I'm not sure it'll fix everything.

Reply:
I have actually seen some more non deterministic issues while having this fix locally, so we should probably solve differently. Thanks